tests/acceptance/test_sections_and_flows.py

```diff
@@ -0,0 +1,503 @@
+                resource_manager.get(FlowKeys.BUTTON_ADD), exact=True
+            ).click()
+        except Exception:
+            page.locator('[test-id="marker-button-add"]').first.click()
+        page.locator(f'[test-id="{MARKER_FLOW_NAME}"]').first.fill("Temp-Flow")
+        # Select start and end sections before applying
+        page.locator(f'[test-id="{MARKER_START_SECTION}"]').first.click()
+        page.keyboard.press("ArrowDown")
+        page.keyboard.press("Enter")
+        page.locator(f'[test-id="{MARKER_END_SECTION}"]').first.click()
+        page.keyboard.press("ArrowDown")
+        page.keyboard.press("Enter")
+        page.locator(f'[test-id="{MARKER_DIALOG_APPLY}"]').first.click()
+        table = page.locator('[test-id="marker-flow-table"]').first
+        table.wait_for(state="visible")
+        # Wait until the table contains the newly created flow
+        deadline_temp = time.time() + ACCEPTANCE_TEST_WAIT_TIMEOUT
+        while time.time() < deadline_temp:
+            try:
+                if "Temp-Flow" in table.inner_text():
+                    break
+            except Exception:
+                pass
+            time.sleep(0.1)
+        assert "Temp-Flow" in table.inner_text()
+        row = table.locator("tbody tr").filter(has_text="Temp-Flow").first
+        row.click()
+        page.locator('[test-id="marker-button-remove"]').first.click()
+        time.sleep(0.2)
+        assert "Temp-Flow" not in table.inner_text()
+
+        # Generate flows from sections and assert at least two are created
+        page.locator('[test-id="marker-button-generate"]').first.click()
+        deadline_gen = time.time() + ACCEPTANCE_TEST_WAIT_TIMEOUT
+        matched_texts: list[str] = []
+        last_texts: list[str] = []
+        while time.time() < deadline_gen:
+            try:
+                rows = table.locator("tbody tr")
+                last_texts = rows.all_text_contents() if rows.count() else []
+                matched_texts = [t for t in last_texts if all(n in t for n in names)]
+                if len(matched_texts) >= 2:
+                    break
+            except Exception:
+                pass
+            time.sleep(0.1)
+        assert (
+            len(matched_texts) >= 2
+        ), f"Expected >=2 generated flows with {names}, got {len(matched_texts)}: {last_texts}"  # noqa
+
+        # Save project and compare with reference
+        page.locator('[test-id="marker-project-save-as"]').first.click()
+        page.locator(f'[test-id="{MARKER_DIALOG_APPLY}"]').first.wait_for(
+            state="visible"
+        )
+        dir_label = resource_manager.get(FileChooserDialogKeys.LABEL_DIRECTORY)
+        page.get_by_label(dir_label, exact=True).fill(str(test_data_tmp_dir))
+        page.locator(f'[test-id="{MARKER_FILENAME}"]').first.fill("test_name")
+        page.locator(f'[test-id="{MARKER_DIALOG_APPLY}"]').first.click()
+        saved_path = test_data_tmp_dir / "test_name.otconfig"
```

Comment:
Extract a function to compare the saved file with a reference file. We need it in other tests, too.